packages/core-p2p/lib/monitor.js

```diff
@@ -15,54 +15,62 @@ const Peer = require('./peer')
 const guard = require('./guard')
 const networkState = require('./utils/network-state')
 
-module.exports = class Monitor {
+const checkDNS = require('./utils/check-dns')
+const checkNTP = require('./utils/check-ntp')
+
+class Monitor {
   /**
    * @constructor
-   * @param  {PeerManager} manager
    * @throws {Error} If no seed peers
    */
-  constructor (manager) {
-    this.manager = manager
-    this.config = config
+  constructor () {
     this.peers = {}
-    this.guard = guard.init(this)
-    this.startForgers = moment().add(this.config.peers.coldStart || 30, 'seconds')
+    this.startForgers = moment().add(config.peers.coldStart || 30, 'seconds')
   }
 
   /**
    * Method to run on startup.
-   * @param {Boolean} networkStart
+   * @param {Object} config
    */
-  async start (networkStart = false) {
+  async start (config) {
+    this.config = config
+
+    await this.__checkDNSConnectivity(config.dns)
+    await this.__checkNTPConnectivity(config.ntp)
+
+    this.guard = guard.init(this)
+
     this.__filterPeers()
 
-    if (!networkStart) {
+    if (!config.networkStart) {
       await this.updateNetworkStatus()
     }
+
+    return this
   }
 
   /**
    * Update network status (currently only peers are updated).
    * @return {Promise}
    */
-  async updateNetworkStatus (fast = false) {
+  async updateNetworkStatus () {
     try {
       // TODO: for tests that involve peers we need to sync them
       if (process.env.ARK_ENV !== 'test') {
         await this.discoverPeers()
         await this.cleanPeers()
       }
 
-      if (Object.keys(this.peers).length < this.config.peers.list.length - 1 && process.env.ARK_ENV !== 'test') {
-        this.config.peers.list
+      if (Object.keys(this.peers).length < config.peers.list.length - 1 && process.env.ARK_ENV !== 'test') {
```

Comment:
Shouldn't this be `this.config`? I don't see config passed into updateNetworkStatus, or declared outside of the class.

Reply:
`config` is the `container.resolvePlugin('config')`

Reply:
I guess i needed sleep, I looked for it at the top and didn't see it 🤦‍♂️